RFD.md

```diff
@@ -0,0 +1,267 @@
+
+Users (clients) will be assigned one of these two roles in a hard-coded map for simplicity. In a real application, this assignment would be handled by an authorization server with groups and permissions.
+
+#### GRPC
+
+The API will be a gRPC server that handles messages from the CLI. Protocol Buffers (protobuf) will be used to define and serialize the methods and messages, which will provide consistency between the gRPC server and client.
+
+The following proto interface will be used:
+```proto
+// Service definition for the Remote Process Service (RPS)
+service RpsService {
+  // Starts a new job with the given command.
+  rpc Start(StartRequest) returns (StartResponse);
+
+  // Gets the status of a specific job.
+  rpc Status(StatusRequest) returns (StatusResponse);
+
+  // Streams the output of a specific job.
+  rpc Output(OutputRequest) returns (stream OutputResponse);
+
+  // Stops a running job.
+  rpc Stop(StopRequest) returns (StopResponse);
+}
+
+// Request message for starting a job.
+message StartRequest {
+  string command = 1; // The command to execute for the job.
+}
+
+// Response message for starting a job.
+message StartResponse {
+  string job_id = 1; // The unique identifier (UUID) for the started job.
+  string error_message = 2; // Error message if the job failed to start.
+}
+
+// Enum representing the possible statuses of a job.
+enum JobStatus {
+  UNKNOWN = 0;
+  RUNNING = 1;
+  STOPPED = 2;
+  ERROR = 3; // Represents a job that terminated with an error.
+}
+
+// Request message for getting the status of a job.
+message StatusRequest {
+  string job_id = 1; // The unique identifier (UUID) of the job.
+}
+
+// Response message containing the status of a job.
+message StatusResponse {
+  string job_id = 1;                      // The unique identifier (UUID) of the job.
+  JobStatus status = 2;                   // The current status of the job.
+  google.protobuf.Timestamp start_time = 3; // The time the job was started.
+  string command = 4;                     // The command executed by the job.
+  int32 exit_code = 5;                    // The exit code if the job has stopped or errored. Only meaningful if status is STOPPED or ERROR.
+  string error_message = 6;               // Error message if the job has errored.
+}
+
+// Request message for streaming the output of a job.
+message OutputRequest {
+  string job_id = 1; // The unique identifier (UUID) of the job.
+}
+
+// Response message containing a chunk of job output.
+message OutputResponse {
+  bytes data = 1; // A chunk of the job's stdout or stderr output.
+  string error_message = 2; // Error message if the output streaming failed.
+}
+
+// Request message for stopping a job.
+message StopRequest {
+  string job_id = 1; // The unique identifier (UUID) of the job to stop.
+}
+
+// Response message for stopping a job.
+message StopResponse {
+  bool success = 1; // Whether the job stopping was successful.
```

Comment:
Why is the success flag needed?

If the gRPC endpoint returns no error, wouldn’t that already imply the stop call succeeded? Could you elaborate on scenarios where success = false but no gRPC error is returned?